salt/proxy/junos.py

```diff
@@ -118,10 +120,18 @@ def conn():
 
 def alive(opts):
     '''
-    Return the connection status with the remote device.
+    Validate and return the connection status with the remote device.
 
     .. versionadded:: Oxygen
     '''
+    try:
+        thisproxy['conn'].rpc('<file-list><path>/dev/null</path></file-list>')
+
+    except RpcTimeoutError:
+        thisproxy['conn'].connected = False
```

Comment:
shutdown will get called as soon as alive returns False. May be unnecessary to do so here?